src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
+  isCommentFormVisible: boolean;
+  userSelected: User | null;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  isCommentLoading,
+  postComments,
+  selectedPost,
+  errorsComment,
+  onDeleteComment,
+  onAddComment,
+  isButtonAddLoading,
+  onShowCommentFormChange,
+  isCommentFormVisible,
+  userSelected,
+}) => {
+  useEffect(() => {
+    onShowCommentFormChange(false);
+  }, [selectedPost]);
+
+  if (!selectedPost || userSelected?.id !== selectedPost.userId) {
+    return null;
+  }
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+          <h2 data-cy="PostTitle">{`#${selectedPost?.id}: ${selectedPost?.title}`}</h2>
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
+          <p data-cy="PostBody">{selectedPost?.body}</p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+          {isCommentLoading ? (
+            <Loader />
+          ) : (
+            <>
+              {errorsComment ? (
+                <div className="notification is-danger" data-cy="CommentsError">
+                  Something went wrong
+                </div>
+              ) : (
+                <>
+                  {postComments.length === 0 ? (
+                    <p className="title is-4" data-cy="NoCommentsMessage">
+                      No comments yet
+                    </p>
+                  ) : (
+                    <p className="title is-4">Comments:</p>
+                  )}
 
-          <p className="title is-4">Comments:</p>
+                  {postComments.map((comment: Comment) => (
+                    <article
+                      className="message is-small"
+                      data-cy="Comment"
+                      key={comment.id}
+                    >
+                      <div className="message-header">
+                        <a
+                          href={`mailto:${comment.email}`}
+                          data-cy="CommentAuthor"
+                        >
+                          {comment.name}
+                        </a>
+                        <button
+                          data-cy="CommentDelete"
+                          type="button"
+                          className="delete is-small"
+                          aria-label="delete"
+                          onClick={() => onDeleteComment(comment.id)}
```

Comment:
You are using `onDeleteComment(comment.id)` directly in the onClick handler. It's better to wrap this in a function to prevent unnecessary renders. You can do something like this: `onClick={() => handleDeleteComment(comment.id)}` and then define `handleDeleteComment` in your component.